src/passes/OptimizeInstructions.cpp

```diff
@@ -874,14 +882,17 @@ struct OptimizeInstructions : public WalkerPass<PostWalker<OptimizeInstructions,
     auto* left = binary->left;
     auto* right = binary->right;
     if (!Properties::emitsBoolean(left) || !Properties::emitsBoolean(right)) return nullptr;
-    auto leftEffects = EffectAnalyzer(getPassOptions(), left).hasSideEffects();
-    auto rightEffects = EffectAnalyzer(getPassOptions(), right).hasSideEffects();
-    if (leftEffects && rightEffects) return nullptr; // both must execute
-   // canonicalize with side effects, if any, happening on the left
-    if (rightEffects) {
+    auto leftEffects = EffectAnalyzer(getPassOptions(), left);
+    auto rightEffects = EffectAnalyzer(getPassOptions(), right);
+    auto leftSideEffects = leftEffects.hasSideEffects();
```

Comment:
Nit: is `leftHasEffects` clearer? Without looking at the definition it's not immediately clear whether `leftEffects` is the boolen or if it's `leftSideEffects`. Not sure if "has effects" makes that too much better, and it seems kind of a wash because "side effects" mirrors the `hasSideEffects` method name.